tests/integration/cloudformation/api/test_stack_policies.py

```diff
@@ -0,0 +1,696 @@
+import json
+import os
+
+import botocore.exceptions
+import pytest
+import yaml
+
+from localstack.testing.aws.util import is_aws_cloud
+from localstack.utils.files import load_file
+from localstack.utils.strings import short_uid
+from localstack.utils.sync import retry
+
+
+def get_events_canceled_by_policy(cfn_client, stack_name):
+    events = cfn_client.describe_stack_events(StackName=stack_name)["StackEvents"]
+
+    failed_events_by_policy = [
+        event
+        for event in events
+        if "ResourceStatusReason" in event
+        and (
+            "Action denied by stack policy" in event["ResourceStatusReason"]
+            or "Action not allowed by stack policy" in event["ResourceStatusReason"]
+            or "Resource update cancelled" in event["ResourceStatusReason"]
+        )
+    ]
+
+    return failed_events_by_policy
+
+
+def delete_stack_after_process(cfn_client, stack_name):
+    progress_is_finished = False
+    while not progress_is_finished:
+        status = cfn_client.describe_stacks(StackName=stack_name)["Stacks"][0]["StackStatus"]
+        progress_is_finished = "PROGRESS" not in status
+    cfn_client.delete_stack(StackName=stack_name)
```

Comment:
What was the reason for adding this waiting loop? Doesn't AWS allow deleting a stack while it's in a progress state?

Reply:
Right, if the stack update fails due to the stack policy, it enters in the state "UPDATE_ROLLBACK_IN_PROGRESS" (or something similar) and the stack cannot be deleted until the process is done.